cmd/apply/cmdapply.go

```diff
@@ -114,10 +114,16 @@ func (r *ApplyRunner) RunE(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
+	namespace, enforceNamespace, err := r.factory.ToRawKubeConfigLoader().Namespace()
+	if err != nil {
+		return err
+	}
+
 	var reader manifestreader.ManifestReader
 	readerOptions := manifestreader.ReaderOptions{
-		Factory:   r.factory,
-		Namespace: metav1.NamespaceDefault,
+		Factory:          r.factory,
+		Namespace:        namespace,
+		EnforceNamespace: enforceNamespace,
```

Comment:
Are we using the `EnforceNamespace` field anywhere?

Reply:
The goal is to use it the same way here as it is used in kubectl. So the if the value of the namespace is coming from the context, the value of `EnforceNamespace` will be `false`, while if the namespace is set with the `--namespace` flag, the `EnforceNamespace` value will be `true`.
The value here is used in the `setNamespaces` function.